honu_test.go

```diff
@@ -310,6 +295,179 @@ func TestUpdate(t *testing.T) {
+	// including tombstones still stored in the database.
+	requireNamespaceLen(t, db, "graveyard", 10)
+	requireGraveyardLen(t, db, "graveyard", 20)
+	requireDatabaseLen(t, db, 20)
+
+	// Sanity check, attempt to get Get all keys and verify tombstones
+	for i, key := range keys {
+		if i%2 == 0 {
+			// This is a tombstone
+			val, err := db.Get(key, options.WithNamespace("graveyard"))
+			require.EqualError(t, err, "not found", "tombstone did not return a not found error")
+			require.Nil(t, val, "tombstone returned a non nil value")
+
+			obj, err := db.Object(key, options.WithNamespace("graveyard"))
+			require.NoError(t, err, "tombstone did not return an object")
+			require.True(t, obj.Tombstone())
+		} else {
+			// Not a tombstone
+			val, err := db.Get(key, options.WithNamespace("graveyard"))
+			require.NoError(t, err, "a live object returned error on get")
+			require.Len(t, val, 256)
+
+			obj, err := db.Object(key, options.WithNamespace("graveyard"))
+			require.NoError(t, err, "live object did not return an object")
+			require.False(t, obj.Tombstone())
+		}
+	}
+
+	// "Resurrect" every 4th tombstone and give it a new value
+	for i, key := range keys {
+		if i%4 == 0 {
+			db.Put(key, randomData(192), options.WithNamespace("graveyard"))
+		}
+	}
+
+	// Ensure that the iterator returns 15 items but that there are still 20 objects
+	// including tombstones still stored in the database.
+	requireNamespaceLen(t, db, "graveyard", 15)
+	requireGraveyardLen(t, db, "graveyard", 20)
+	requireDatabaseLen(t, db, 20)
+
+	// Sanity check, attempt to get Get all keys and verify tombstones and undead keys
+	for i, key := range keys {
+		if i%2 == 0 {
+			if i%4 == 0 {
+				// This is an undead version
+				val, err := db.Get(key, options.WithNamespace("graveyard"))
+				require.NoError(t, err, "undead object returned error on get")
+				require.Len(t, val, 192)
+
+				obj, err := db.Object(key, options.WithNamespace("graveyard"))
+				require.NoError(t, err, "undead object did not return an object")
+				require.False(t, obj.Tombstone())
+			} else {
+				// This is a tombstone
+				val, err := db.Get(key, options.WithNamespace("graveyard"))
+				require.EqualError(t, err, "not found", "tombstone did not return a not found error")
+				require.Nil(t, val, "tombstone returned a non nil value")
+
+				obj, err := db.Object(key, options.WithNamespace("graveyard"))
+				require.NoError(t, err, "tombstone did not return an object")
+				require.True(t, obj.Tombstone())
+			}
+		} else {
+			// Not a tombstone
+			val, err := db.Get(key, options.WithNamespace("graveyard"))
+			require.NoError(t, err, "a live object returned error on get")
+			require.Len(t, val, 256)
+
+			obj, err := db.Object(key, options.WithNamespace("graveyard"))
+			require.NoError(t, err, "live object did not return an object")
+			require.False(t, obj.Tombstone())
+		}
+	}
+}
+
+func TestTombstonesMultipleNamespaces(t *testing.T) {
```

Comment:
@pdeziel I'm hoping this resolves SC-3389; let me know if the logic doesn't look right to you.